ui/app/components/clients/date-range.ts

```diff
@@ -62,6 +66,21 @@ export default class ClientsDateRangeComponent extends Component<Args> {
     return parseAPITimestamp(isoTimestamp, 'MMMM yyyy');
   };
 
+  get historicalBillingPeriods() {
+    const count = this.args.retentionMonths / 12;
+    const periods: string[] = [];
+
+    // for each historical billing period, start_date will be billing_start_date - (12 * multiple)
+    for (let i = 1; i <= count; i++) {
+      const startDate = new Date(this.args.billingStartTime);
```

Comment:
Just a heads up about a subtle issue here using `new Date()`. Even though we're passing a UTC timestamp, javascript Date objects are automatically converted to local time which can introduce inconsistencies when doing date math. 

For example, `new Date("2024-05-08T00:00:00Z")` in a Pacific time zone becomes May 7, 2024 at 5:00 PM, and subtracting months from that can lead to off-by-one-day errors or unexpected rollovers.

To keep everything in UTC and avoid local time zone drift, you could use Date.UTC(...) combined with getUTCFullYear(), getUTCMonth(), etc. to ensure billing periods are calculated consistently.

Let me know if you have more questions! It might make sense to make this a util.